stl/inc/xstring

```diff
@@ -2240,11 +2242,17 @@ public:
         return _Result;
     }
 
-    _CONSTEXPR20 bool _Large_string_engaged() const noexcept {
-        return _BUF_SIZE <= _Myres;
+    _NODISCARD static _CONSTEXPR20 bool _Large_string_engaged(size_type _Capacity) noexcept {
+        // determines whether large mode is used for this capacity
+        // has no implication on largeness of string's size
+        return _Capacity > _SMALL_STRING_CAPACITY;
     }

```

Comment:
This is a philosophical point, but I strongly believe that function overloading should be used only when the programmer "doesn't care" which overload is selected. For example, overloading `to_string(short)` and `to_string(unsigned long long)` is totally fine, because the programmer has already chosen a data type, and now they want a `string`, and they don't care how it happens.

Here, there is a *major* difference between the `static` and non-`static` overloads of `_Large_string_engaged()`, which is dangerous. The `static` doesn't prevent mistakes, because `static` functions can be called with non-`static` syntax.

`_Entails_large_string()` has the exact same behavior, but a much better name (the different word order is critical); I believe we should use it uniformly.